clients/ts/signalr/src/HttpClient.ts

```diff
@@ -16,12 +16,14 @@ export interface HttpRequest {
 }
 
 export class HttpResponse {
+    constructor(statusCode: number);
+    constructor(statusCode: number, statusText: string);
     constructor(statusCode: number, statusText: string, content: string);
     constructor(statusCode: number, statusText: string, content: ArrayBuffer);
     constructor(
         public readonly statusCode: number,
-        public readonly statusText: string,
-        public readonly content: string | ArrayBuffer) {
+        public readonly statusText?: string,
+        public readonly content?: string | ArrayBuffer) {
```

Comment:
What is this change for? I'm not 100% up to speed on the intricacies of TS overloading.

Reply:
This is just how you make overloads, you add new signatures and you make the actual function definition accept optional parameters as needed so that all the signature combinations can be passed in properly, since JavaScript doesn't **actually** have overloads.

See "Overloads" in https://www.typescriptlang.org/docs/handbook/functions.html